pkg/util/http.go

```diff
@@ -19,6 +20,22 @@ import (
 
 const messageSizeLargerErrFmt = "received message larger than max (%d vs %d)"
 
+// BasicAuth configures basic authentication for HTTP clients.
+type BasicAuth struct {
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+}
+
+func (b *BasicAuth) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.StringVar(&b.Username, prefix+"username", "", "Username that will be set basic authentication in requests.")
+	f.StringVar(&b.Password, prefix+"password", "", "Password that will be set for basic authentication in requests.")
```

Comment:
I'm wondering if the following description may be more clear:
`HTTP Basic authentication username. It overrides the username set in the URL (if any).`

_Same comment applies to the password._

Reply:
I agree that's much clearer. I'll update it.